### DISCUSSION TITLE
alist v3.12.0，docker容器一段时间后自动关闭且不会重启

### DISCUSSION THREAD

**Discussion by JohnDoeCN** [AlistGo/alist#3605] [Q&A] [answered]
威联通Container Station，在dockerhub上搜索alist，选择3.12.0版本安装
添加了百度网盘和阿里网盘，webdav读取写入正常工作，但每次运行半小时左右，容器就自动关闭，且不会自动重启。
这个过程中我使用webdav连接alist，从两个网盘读写数据。
请大佬帮忙看看原因，谢谢。
容器控制台相关信息：
INFO[2023-02-24 16:12:58] success load storage: [/dupan], driver: [BaiduNetdisk] 
INFO[2023-02-24 16:12:59] success load storage: [/aliwangpan], driver: [AliyundriveOpen] 
2023/02/24 16:33:37.243945 ERROR RESTY Post "https://open.aliyundrive.com/adrive/v1.0/openFile/list": dial tcp: lookup open.aliyundrive.com on 10.0.3.1:53: read udp 10.0.3.2:60961->10.0.3.1:53: i/o timeout, Attempt 1
2023/02/24 16:33:47.347584 ERROR RESTY Post "https://open.aliyundrive.com/adrive/v1.0/openFile/list": dial tcp: lookup open.aliyundrive.com on 10.0.3.1:53: read udp 10.0.3.2:60799->10.0.3.1:53: i/o timeout, Attempt 2
2023/02/24 16:35:55.413840 ERROR RESTY Post "https://open.aliyundrive.com/adrive/v1.0/openFile/list": dial tcp: lookup open.aliyundrive.com on 10.0.3.1:53: read udp 10.0.3.2:35371->10.0.3.1:53: i/o timeout, Attempt 1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x125e890]

goroutine 94 [running]:
github.com/alist-org/alist/v3/drivers/aliyundrive_open.(*AliyundriveOpen).Init.func1()
        /app/drivers/aliyundrive_open/driver.go:52 +0x30
github.com/alist-org/alist/v3/pkg/cron.(*Cron).Do.func1()
        /app/pkg/cron/cron.go:24 +0x8e
created by github.com/alist-org/alist/v3/pkg/cron.(*Cron).Do
        /app/pkg/cron/cron.go:18 +0x72

![image](https://user-images.githubusercontent.com/47625783/221328171-efae7c69-fe56-4158-8e44-566e0fbadcf3.png)
![image](https://user-images.githubusercontent.com/47625783/221328205-19792748-71be-4092-9b81-cfd9d25bdf1c.png)
![image](https://user-images.githubusercontent.com/47625783/221328229-d733b908-9c12-4877-abc7-1eecce59deb9.png)
![image](https://user-images.githubusercontent.com/47625783/221328236-f22dc98f-8dfb-49e7-a74b-6c5173e5152b.png)
![image](https://user-images.githubusercontent.com/47625783/221328249-c314e4ab-1da1-4961-9917-d25e42ec25df.png)

**Comment by miniidc** [comment]
不光容器 我的也会自动关闭，需要启动命令才可以使用。

**Reply by xupengju110** [reply]
我的win系统也是自动关闭，需要手动启动

**Comment by bed-man** [comment]
电脑端也会自动关闭且不会重启，，，怎么解决啊？

**Comment by mzoux** [comment]
我的也是，手动安装和一键安装都不行，都会掉进程。

**Reply by JohnDoeCN** [reply]
你们也是3.12.0吗？

**Reply by mzoux** [reply]
是啊

**Reply by JohnDoeCN** [reply]
你也用阿里网盘open？我看报错都跟这个open接口有关

**Comment by xcf13363175** [comment]
更新到3.12用open，一键安装同样掉进程。

**Reply by JohnDoeCN** [reply]
基本确定是阿里云盘open的问题，停用了阿里的存储，就不会掉了

**Comment by Sa1koro** [comment]
alist报错一样，systemd如下
```shell
alist[6750]: panic: runtime error: invalid memory address or nil pointer dereference
alist[6750]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x12a6990]
alist[6750]: goroutine 89 [running]:
alist[6750]: github.com/alist-org/alist/v3/drivers/aliyundrive_open.(*AliyundriveOpen).Init.func1()
alist[6750]:         /home/runner/work/alist/alist/drivers/aliyundrive_open/driver.go:52 +0x30
alist[6750]: github.com/alist-org/alist/v3/pkg/cron.(*Cron).Do.func1()
alist[6750]:         /home/runner/work/alist/alist/pkg/cron/cron.go:24 +0x8e
alist[6750]: created by github.com/alist-org/alist/v3/pkg/cron.(*Cron).Do
alist[6750]:         /home/runner/work/alist/alist/pkg/cron/cron.go:18 +0x72
systemd[1]: alist.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
systemd[1]: alist.service: Unit entered failed state.
systemd[1]: alist.service: Failed with result 'exit-code'.
```
所以怎么才能让服务意外关闭后 自动重启呀？

**Comment by allSaved** [comment]
我的3.12.2也是，windows环境下，只挂载了阿里云open。每次启动几小时后就崩溃退出，显示：panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x10 pc=0x1857568]

**Comment by JohnDoeCN** [comment, marked as answer]
更新到3.12.2，已经没有问题了。
https://github.com/alist-org/alist/issues/3617
